src/planar_rigid_body_visualizer.py

```diff
@@ -0,0 +1,296 @@
+                            # points around the sphere surface in the view plane.
+                            center = geom.getPoints()
+                            sample_pts = np.arange(0., 2.*math.pi, 0.25)
+                            patch = np.vstack([math.cos(pt)*self.Tview[0, 0:3] + math.sin(pt)*self.Tview[1, 0:3] for pt in sample_pts])
+                            patch = np.transpose(patch)
+                            patch *= geom.radius
+                        else:
+                            # Other geometry types will usually return their
+                            # bounding box, which is good enough for basic
+                            # visualization. (This could be improved for 
+                            # capsules?)
+                            patch = geom.getPoints()
+                    # Convert to homogenous coords and move out of body frame
+                    patch = np.vstack((patch, np.ones((1, patch.shape[1]))))
+                    patch = np.dot(element_local_tf, patch)
+                    # Project into 2D
+                    patch = np.dot(self.Tview, patch)
+
+                    # Take convhull of resulting points
+                    if patch.shape[1] > 3:
+                        hull = sp.spatial.ConvexHull(np.transpose(patch[0:2, :]))
+                        patch = np.transpose(np.vstack([patch[:, v] for v in hull.vertices]))
+                    this_body_patches.append(patch)
+                    if use_random_colors:
+                        this_body_colors.append(next(color))
+                    else:
+                        this_body_colors.append(element.getMaterial())
+
+            self.viewPatches.append(this_body_patches)
+            self.viewPatchColors.append(this_body_colors)
+
+
+    def getViewPatches(self, body_i, tf):
+        ''' Pulls out the view patch verts for the given body index after applying
+            the appropriate TF '''
+        projected_tf = np.dot(np.dot(self.Tview, tf), self.Tview_pinv)
+        transformed_patches = [np.dot(projected_tf, patch)[0:2] for patch in self.viewPatches[body_i]]
+        colors = self.viewPatchColors[body_i]
+        return (transformed_patches, colors)
+
+    def draw(self, context):
+        ''' Evalutates the robot state and draws it. '''
+        positions = self.EvalVectorInput(context, 0).get_value()[0:self.rbtree.get_num_positions()]
+
+        kinsol = self.rbtree.doKinematics(positions)
+
+        body_fill_index = 0
+        for body_i in range(self.rbtree.get_num_bodies()-1):
+            tf = self.rbtree.relativeTransform(kinsol, 0, body_i+1)
+            viewPatches, _ = self.getViewPatches(body_i, tf)
+            for patch in viewPatches:
+                self.body_fill_list[body_fill_index].get_path().vertices[:, :] = np.transpose(patch)
+                body_fill_index += 1
+
+if __name__ == "__main__":
+    # Usage demo: load a URDF, rig it up with a constant torque input, and draw it.
```

Comment:
add it to the CMakeLists.txt as a unit test?  (ideally with small sim durations, so we know if it breaks, but don't chew up frivolous cycles on circleci)